### ISSUE TITLE
2 changes triggers re-renders on initial mount

### ISSUE THREAD

**Issue by gabrielbull** [th3rdwave/react-native-safe-area-context#13]
### Current behaviour

Initial mounting of the SafeAreaProvider triggers 2 updates to the insets.

For instance, on an iPhone 11 Pro, we get these three updates on the initial mounting of the provider:

##### Render 1:
```
{left: 0, top: 44, right: 0, bottom: 0}
```

##### Render 2:
```
{left: 0, top: 0, right: 0, bottom: 0}
```

##### Render 3:
```
{left: 0, top: 44, right: 0, bottom: 34}
```

### Expected behaviour
Expected behaviour would be one render with the final value of render 3.

### Code to reproduce

```jsx
import React from 'react'
import { SafeAreaProvider, SafeAreaConsumer } from 'react-native-safe-area-context'

const App = () => (
  <SafeAreaProvider>
    <SafeAreaConsumer>
      {insets => {
        console.log(insets) // Gets called 3 times
        return null
      }}
    </SafeAreaConsumer>
  </SafeAreaProvider>
)

export default App
```

**Comment by janicduplessis** [comment]
Does that happen in dev when reloading JS or do you observe the same behaviour in release / initial render (killing app and restarting completely)?

**Comment by gabrielbull** [comment]
Yes it does. It isn't consistent though, I've had 1, 2, 3 and up to 4 re-renders on startup.

Here's the code altered to show the results on a release build:

```jsx
import React, {useRef} from 'react'
import {View, Text} from 'react-native'
import { SafeAreaProvider, SafeAreaConsumer } from 'react-native-safe-area-context'

const App = () => {
  const insetsRef = useRef([])
  return (
    <SafeAreaProvider>
      <SafeAreaConsumer>
        {insets => {
          if (insets) insetsRef.current.push(insets)
          return (
            <View style={{marginTop: 100}}>
              {insetsRef.current.map((insets, index) => <Text key={index}>{`{top: ${insets.top}, bottom: ${insets.bottom}, left: ${insets.left}, right: ${insets.right}}`}</Text>)}
            </View>
          )
        }}
      </SafeAreaConsumer>
    </SafeAreaProvider>
  )
}

export default App
```

![RPReplay_Final1570459875 2019-10-07 10_55_46](https://user-images.githubusercontent.com/671923/66322986-0fb82700-e8f1-11e9-9416-36a09d8cf7a5.gif)

En passant, salut 👋🏻, on est presque voisin.

**Comment by satya164** [comment]
I can repro this by reloading the app in dev. It doesn't happen every time, but happens pretty often.

**Comment by janicduplessis** [comment]
Ok I figured out what is going on, the insets event is triggered before RN set the view size so it causes wrong insets to be sent since view size is 0. eb234c8db338ebdfaee47c4b90c31b7bfae2c33f should fix it. Released as 0.4.1.
